src/Cms/AppPlugins.php

```diff
@@ -225,6 +228,51 @@ protected function extendFileMethods(array $methods): array
         return $this->extensions['fileMethods'] = File::$methods = array_merge(File::$methods, $methods);
     }
 
+    /**
+     * Registers additional custom file types and mimes
+     *
+     * @param array $fileTypes
+     * @return array
+     */
+    protected function extendFileTypes(array $fileTypes): array
+    {
+        // normalize array
+        foreach ($fileTypes as $ext => $file) {
+            $extension = $file['extension'] ?? $ext;
+            $type      = $file['type'] ?? null;
+            $mime      = $file['mime'] ?? null;
+            $resizable = $file['resizable'] ?? false;
+            $viewable  = $file['viewable'] ?? false;
+
+            if ($type !== null && in_array($extension, F::$types[$type]) === false) {
+                F::$types[$type][] = $extension;
+            }
+
+            if ($mime !== null) {
+                if (array_key_exists($extension, Mime::$types) == true) {
+                    Mime::$types[$extension] = array_merge((array)Mime::$types[$extension], (array)$mime);
```

Comment:
Since we do not know whether the current value and the sent values are strings or arrays, we cast both variables to the array.

Reply:
Good point!

One minor change to ensure that the same type is only included once:

```suggestion
                    Mime::$types[$extension] = array_unique(array_merge((array)Mime::$types[$extension], (array)$mime));
```